modules/javafx.base/src/main/java/com/sun/javafx/binding/Subscription.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javafx.binding;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+
+import javafx.beans.InvalidationListener;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+
+/**
+ * A subscription encapsulates how to cancel it without having
+ * to keep track of how it was created.
+ *
+ * <p>For example:
+ *
+ * <p>{@code Subscription s = property.subscribe(System.out::println)}
+ *
+ * <p>The function passed in to {@code subscribe} does not need to be stored
+ * in order to clean up the subscription later.
+ */
+@FunctionalInterface
+public interface Subscription {
+
+    /**
+     * An empty subscription. Does nothing when cancelled.
+     */
+    static final Subscription EMPTY = () -> {};
+
+    /**
+     * Cancels this subscription.
+     */
+    void unsubscribe();
```

Comment:
to me `unsubscribe` feels not natural, I'm not unsubscribing a `Subscription` but I'm eg disposing it, hence to me `dispose` would feel better but I'm not a native speaker.

Another pattern I see often is that I store a subscription in a field and recreate it under certain conditions:

```java
private Subscription mySubscription;

void refresh() {
   if( this.mySubscription != null ) {
     this.mySubscription.dispos();
   }
   this.mySubscription = ....
}
```

if Subscription would provide

```java 
static void dispose(Subscription subscription) {
  if( subscription != null ) {
    subscription.dispose();
  }
}
```

this would safe the user a null-check.

Reply:
This API is not going to be public for now, so we still have some time to work out the details.  Another option would be to use `cancel` instead of `unsubscribe` or `dispose` (an internal class in the JDK uses `cancel`, reactfx uses `unsubcribe`).  

    Subscription.dispose(subscription);

I'm not sure how I feel about a helper method like that to do a `null` check; I feel that `null` checks are the callers responsibility (in almost all situations).  It is also not really OO, it feels like `void *x = malloc(1000)` and `free(x)` instead of getting an object and calling `free` on the object.